tests/code-base/test_code_base.py

```diff
@@ -32,6 +31,11 @@ def setUp(self):
         open(p2 / "quux.h", mode="w").close()
         open(p2 / "README.md", mode="w").close()
 
+    @classmethod
+    def tearDownClass(self):
+        self.tmp1.cleanup()
```

Comment:
Yeah this kind of thing is what makes it a bit clunky to me - I know that `tearDown` methods are meant to be called under all circumstances, pass or fail, but it makes it hard to validate behavior

Reply:
I don't disagree, and I'm not opposed to finding a better way to rewrite these tests.  But I'd like to defer it until a future PR, since you're not opposed -- I want to make sure that we actually did fix the issue, before spending additional time refactoring and/or rewriting tests.